src/components/Nav/Menu/SubMenu.tsx

```diff
@@ -0,0 +1,162 @@
+ * @param activeSection - English label of the active section for event tracking
+ * @param onClose - Function to close the menu
+ * @returns The JSX element representing the menu content.
+ */
+const SubMenu = ({ lvl, items, activeSection, onClose }: LvlContentProps) => {
+  const { asPath, locale, menuColors, menuVariants, PADDING } = useSubMenu()
+
+  if (lvl > 3) return null
+
+  const templateColumns = `repeat(${4 - lvl}, 1fr)`
+
+  return (
+    <NavigationMenu.Sub orientation="vertical" asChild>
+      <AnimatePresence>
+        <Grid
+          as={motion.div}
+          variants={menuVariants}
+          initial="closed"
+          animate="open"
+          exit="closed"
+          w="full"
+          h="full"
+          gridTemplateColumns={templateColumns}
+        >
+          <NavigationMenu.List asChild>
+            <UnorderedList listStyleType="none" p={PADDING / 2} m="0">
+              {items.map((item) => {
+                const { label, description, icon, ...action } = item
+                const subItems = action.items || []
+                const isLink = "href" in action
+                const isActivePage = isLink && cleanPath(asPath) === action.href
+                const activeStyles = {
+                  outline: "none",
+                  rounded: "md",
+                  "p, svg": { color: menuColors.highlight },
+                  bg: menuColors.lvl[lvl].activeBackground,
+                  boxShadow: "none",
+                }
+                const buttonProps: ButtonProps = {
+                  color: menuColors.body,
+                  leftIcon: lvl === 1 && icon ? <Icon as={icon} /> : undefined,
+                  rightIcon: isLink ? undefined : <NextChevron />,
+                  position: "relative",
+                  w: "full",
+                  me: -PADDING,
+                  sx: {
+                    "span:first-of-type": { m: 0, me: 4 }, // Spacing for icon
+                  },
+                  py: PADDING,
+                  bg: isActivePage
+                    ? menuColors.lvl[lvl].activeBackground
+                    : "none",
+                  _hover: activeStyles,
+                  _focus: activeStyles,
+                  variant: "ghost",
+                }
+                return (
+                  <NavigationMenu.Item key={label} asChild>
+                    <ListItem
+                      mb={PADDING / 2}
+                      _last={{ mb: 0 }}
+                      sx={{
+                        '&:has(button[data-state="open"])': {
+                          roundedStart: "md",
+                          roundedEnd: "none",
+                          bg: menuColors.lvl[lvl].activeBackground,
+                          me: -PADDING,
+                          pe: PADDING,
+                        },
+                      }}
+                    >
+                      {isLink ? (
+                        <NextLink href={action.href!} passHref legacyBehavior>
```

Comment:
@pettinarip fyi, in case you were wondering why I took this approach: https://www.radix-ui.com/primitives/docs/components/navigation-menu#with-client-side-routing